modules/osdk-building-ansible-operator.adoc

```diff
@@ -0,0 +1,377 @@
+can be useful in certain cases. For example, the `cert-manager` operator is
+often deployed with cluster-scoped permissions and watches so that it can manage
+issuing certificates for an entire cluster.
++
+To create your `memcached-operator` project to be cluster-scoped, use the
+following command:
++
+----
+$ operator-sdk new memcached-operator \
+    --cluster-scoped \
+    --api-version=cache.example.com/v1alpha1 \
+    --kind=Memcached \
+    --type=ansible
+----
++
+Using the `--cluster-scoped` flag scaffolds the new Operator with the following
+modifications:
++
+--
+* `deploy/operator.yaml`: Set `WATCH_NAMESPACE=""` instead of setting it to the
+Pod's namespace.
+* `deploy/role.yaml`: Use `ClusterRole` instead of `Role`.
+* `deploy/role_binding.yaml`:
+** Use `ClusterRoleBinding` instead of `RoleBinding`.
+** Set the subject namespace to `REPLACE_NAMESPACE`. This must be changed to the
+namespace in which the Operator is deployed.
+--
+
+. *Customize the Operator logic.*
++
+For this example, the `memcached-operator` executes the following reconciliation
+logic for each `Memcached` Custom Resource (CR):
++
+--
+* Create a `memcached` Deployment if it does not exist.
+* Ensure that the Deployment size is the same as specified by the `Memcached` CR.
+--
++
+By default, the `memcached-operator` watches `Memcached` resource events as
+shown in the `watches.yaml` file and executes the Ansible role `Memcached`:
++
+[source,yaml]
+----
+- version: v1alpha1
+  group: cache.example.com
+  kind: Memcached
+----
++
+You can optionally customize the following logic in the `watches.yaml` file:
+
+.. Specifying a `role` option configures the Operator to use this specified path
+when launching `ansible-runner` with an Ansible role. By default, the new
+command fills in an absolute path to where your role should go:
++
+[source,yaml]
+----
+- version: v1alpha1
+  group: cache.example.com
+  kind: Memcached
+  role: /opt/ansible/roles/memcached
+----
+
+.. Specifying a `playbook` option in the `watches.yaml` file configures the
+Operator to use this specified path when launching `ansible-runner` with an
+Ansible playbook:
++
+[source,yaml]
+----
+- version: v1alpha1
+  group: cache.example.com
+  kind: Memcached
+  playbook: /opt/ansible/playbook.yaml
+----
+
+. *Build the Memcached Ansible role.*
++
+Modify the generated Ansible role under the `roles/memcached/` directory. This
+Ansible role controls the logic that is executed when a resource is modified.
+
+.. *Define the `Memcached` spec.*
++
+Defining the spec for an Ansible-based Operator can be done entirely in Ansible.
+The Ansible Operator passes all key-value pairs listed in the CR spec field
+along to Ansible as
+link:https://docs.ansible.com/ansible/2.5/user_guide/playbooks_variables.html#passing-variables-on-the-command-line[variables].
+The names of all variables in the spec field are converted to snake case by the
```

Comment:
If it's understood by others that's fine, but I had to google what snake case was :)

Reply:
Yeah, I was trying to decide how much to explain that here, but was hoping the "For example" that follows it would be enough. I've added a parenthetical now, though: "(lowercase with an underscore)".